GUITests/Core/Engine.cs

```diff
@@ -0,0 +1,132 @@
+﻿using GUITestsEngine.Utils;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace GUITestsEngine.Core
+{
+    public static class Engine
+    {
+        private static ConcurrentDictionary<Type, MethodSet> methodsToTest = new ConcurrentDictionary<Type, MethodSet>();
+
+        private static ConcurrentDictionary<Type, ConcurrentBag<TestInfo>> testResults = new ConcurrentDictionary<Type, ConcurrentBag<TestInfo>>();
+
+        private static IEnumerable<Type> GetClasses()
+        {
+            return Assembly.GetExecutingAssembly().ExportedTypes.Where(t => t.IsClass);
+        }
+
+        private static void QueueClassTests(Type type)
+        {
+            var methodSet = new MethodSet(type);
+
+            if (methodSet.TestsCount > 0)
+            {
+                methodsToTest.TryAdd(type, methodSet);
+            }
+        }
+
+        private static void RunCollectedTests()
+        {
+            Parallel.ForEach(methodsToTest.Keys, type =>
+            {
+                testResults.TryAdd(type, new ConcurrentBag<TestInfo>());
+
+                foreach (var testMethod in methodsToTest[type].TestMethods)
+                {
+                    RunTestMethod(type, testMethod);
+                }
+            });
+        }
+
+        private static void RunTestMethod(Type type, MethodInfo method)
+        {
+            var ctor = type.GetConstructor(Type.EmptyTypes);
+
+            if (ctor == null)
+            {
+                throw new FormatException($"{type.Name} must have parameterless constructor");
+            }
+
+            var instance = ctor.Invoke(null);
+
+            var isPassed = false;
+
+            try
+            {
+                method.Invoke(instance, null);
+                isPassed = true;
+            }
+            catch { }
```

Comment:
Why do we need this empty "catch block"?

Reply:
> Why do we need this empty "catch block"?

Not to fall after exception is thrown, but IsPassed was created ass false already, so we don't need to reassign its value here